holoviews/core/data/cudf.py

```diff
@@ -285,7 +286,11 @@ def aggregate(cls, dataset, dimensions, function, **kwargs):
             grouped = reindexed.groupby(cols, sort=False)
             if not hasattr(grouped, agg):
                 raise ValueError(f'{agg} aggregation is not supported on cudf DataFrame.')
-            df = getattr(grouped, agg)().reset_index()
+            numeric_cols = [
+                c for c, d in zip(reindexed.columns, reindexed.dtypes)
+                if is_numeric_dtype(d) and c not in cols
+            ]
+            df = getattr(grouped[numeric_cols], agg)().reset_index()
```

Comment:
No longer discard non-numeric cols; the same things have been done in the panda's Backend. MRE of failing test `holoviews/tests/core/data/test_cudfinterface.py::cuDFInterfaceTests::test_dataset_aggregate_string_types

``` python
import numpy as np
from holoviews import Dataset

Dataset.datatype = ['cuDF']

ds = Dataset({'Gender':['M', 'M'], 'Weight':[20, 10], 'Name':['Peter', 'Matt']},
                     kdims='Gender', vdims=['Weight', 'Name'])
ds.aggregate(['Gender'], np.mean)
```

``` python-traceback
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[2], line 8
      4 Dataset.datatype = ['cuDF']
      6 ds = Dataset({'Gender':['M', 'M'], 'Weight':[20, 10], 'Name':['Peter', 'Matt']},
      7                      kdims='Gender', vdims=['Weight', 'Name'])
----> 8 ds.aggregate(['Gender'], np.mean)

File [~/projects/holoviz/repos/holoviews/holoviews/core/data/__init__.py:196](http://localhost:8888/lab/tree/holoviews/core/data/__init__.py#line=195), in PipelineMeta.pipelined.<locals>.pipelined_fn(*args, **kwargs)
    193     inst._in_method = True
    195 try:
--> 196     result = method_fn(*args, **kwargs)
    197     if PipelineMeta.disable:
    198         return result

File [~/projects/holoviz/repos/holoviews/holoviews/core/data/__init__.py:922](http://localhost:8888/lab/tree/holoviews/core/data/__init__.py#line=921), in Dataset.aggregate(self, dimensions, function, spreadfn, **kwargs)
    919     return self.clone([], kdims=kdims, vdims=vdims, new_type=new_type)
    921 vdims = self.vdims
--> 922 aggregated, dropped = self.interface.aggregate(self, kdims, function, **kwargs)
    923 aggregated = self.interface.unpack_scalar(self, aggregated)
    924 vdims = [vd for vd in vdims if vd not in dropped]

File [~/projects/holoviz/repos/holoviews/holoviews/core/data/cudf.py:288](http://localhost:8888/lab/tree/holoviews/core/data/cudf.py#line=287), in cuDFInterface.aggregate(cls, dataset, dimensions, function, **kwargs)
    286     if not hasattr(grouped, agg):
    287         raise ValueError(f'{agg} aggregation is not supported on cudf DataFrame.')
--> 288     df = getattr(grouped, agg)().reset_index()
    289 else:
    290     agg_map = {'amin': 'min', 'amax': 'max', 'size': 'count'}

File ~/projects/holoviz/repos/holoviews/.pixi/envs/test-gpu/lib/python3.11/site-packages/cudf/core/mixins/mixin_factory.py:11, in _partialmethod.<locals>.wrapper(self, *args2, **kwargs2)
     10 def wrapper(self, *args2, **kwargs2):
---> 11     return method(self, *args1, *args2, **kwargs1, **kwargs2)

File ~/projects/holoviz/repos/holoviews/.pixi/envs/test-gpu/lib/python3.11/site-packages/cudf/core/groupby/groupby.py:759, in GroupBy._reduce(self, op, numeric_only, min_count, *args, **kwargs)
    757 if numeric_only:
    758     return self._reduce_numeric_only(op)
--> 759 return self.agg(op)

File ~/projects/holoviz/repos/holoviews/.pixi/envs/test-gpu/lib/python3.11/site-packages/nvtx/nvtx.py:116, in annotate.__call__.<locals>.inner(*args, **kwargs)
    113 @wraps(func)
    114 def inner(*args, **kwargs):
    115     libnvtx_push_range(self.attributes, self.domain.handle)
--> 116     result = func(*args, **kwargs)
    117     libnvtx_pop_range(self.domain.handle)
    118     return result

File ~/projects/holoviz/repos/holoviews/.pixi/envs/test-gpu/lib/python3.11/site-packages/cudf/core/groupby/groupby.py:631, in GroupBy.agg(self, func)
    621 orig_dtypes = tuple(c.dtype for c in columns)
    623 # Note: When there are no key columns, the below produces
    624 # an Index with float64 dtype, while Pandas returns
    625 # an Index with int64 dtype.
    626 # (GH: 6945)
    627 (
    628     result_columns,
    629     grouped_key_cols,
    630     included_aggregations,
--> 631 ) = self._groupby.aggregate(columns, normalized_aggs)
    633 result_index = self.grouping.keys._from_columns_like_self(
    634     grouped_key_cols,
    635 )
    637 multilevel = _is_multi_agg(func)

File groupby.pyx:192, in cudf._lib.groupby.GroupBy.aggregate()

TypeError: function is not supported for this dtype: mean
```